c/src/ml-api-inference-pipeline.c

```diff
@@ -1777,14 +1777,21 @@ get_app_src_callback (ml_pipeline_common_elem * src_h, void **data)
 {
   ml_pipeline_src_callbacks_s *src_cb = NULL;
   ml_pipeline_element *elem;
+  ml_pipeline *p;
 
+  p = src_h->pipe;
   elem = src_h->element;
+
+  g_mutex_lock (&p->lock);
```

Comment:
What's the op/data protected by this lock in line 1788-1791?

Reply:
This function returns appsrc callback which sets with pipeline lock, so svace-checker returns error when getting cb info without p-lock.

Reply:
The problem is that... ```ml_pipeline_common_elem``` has ```{ callback_info_s , ml_pipeline }```
and ```ml_pipeline->lock``` is protecting ```callback_info_s->*```.

this lock should protect elements in ml_pipeline, not others in ml_pipeline_common_elem.
If it really needs to protect  callback_info_s, it means the lock is really messed up.

Reply:
The lock needs to declare (in doxygen) which are protected by the lock.